### PR TITLE
Kraken cleaning

### PR DESCRIPTION
**Pull request by goodboy**
Follow up to #214 and includes i think at least fix for #290.

### CODE REVIEW

File: piker/brokers/kraken.py [hunk 1]
```diff
@@ -124,11 +124,11 @@ class Trade(BaseModel):
     Trade class that helps parse and validate ownTrades stream
 
     '''
-    reqid: str # kraken order transaction id
-    action: str # buy or sell
-    price: str # price of asset
-    size: str # vol of asset
-    broker_time: str # e.g GTC, GTD
+    reqid: str  # kraken order transaction id
```
Comment (goodboy) [inline piker/brokers/kraken.py:127]: @iamzoltan fyi this is all the whitespace stuff you're missing bc no linter 😉

File: piker/brokers/kraken.py [hunk 2]
```diff
@@ -616,42 +617,41 @@ async def handle_order_requests(
...
-            except AssertionError:
-                log.error(f'Order cancel was not successful')
+            # Check to make sure there was no error returned by
+            # the kraken endpoint. Assert one order was cancelled
+            assert resp['error'] == []
+            assert resp['result']['count'] == 1
+
+            # Check to make sure the cancellation is NOT pending,
+            # then send the confirmation to the ems order stream
+            pending = resp['result'].get('pending')
```
Comment (goodboy) [inline piker/brokers/kraken.py:627]: please be sure this logic looks right, it's not clear to me at all why you assert things then catch the error around another block that has nothing to do with the assert. this is just generally non-pythonic to catch and handle asserts; the whole point is they raise.

File: piker/brokers/kraken.py [hunk 3]
```diff
@@ -772,13 +756,21 @@ async def subscribe(ws: wsproto.WSConnection, token: str):
...
                             filled=float(trade.size),
                             reason='Order filled by kraken',
-                            # remaining='' # TODO: not sure what to do here.
                             broker_details={
                                 'name': 'kraken',
                                 'broker_time': trade.broker_time
-                            }
+                            },
+
+                            # TODO: figure out if kraken gives a count
```
Comment (iamzoltan) [inline piker/brokers/kraken.py:764]: looks good so far.

I think kraken will send the individual trades made, even if the order was complete over several trades. does this mean we need to handle the decrement ourselves?

File: piker/clearing/_messages.py [hunk 4]
```diff
@@ -184,7 +184,7 @@ class BrokerdStatus(BaseModel):
     # {
     #   'submitted',
     #   'cancelled',
-    #   'executed',
+    #   'filled',
```
Comment (iamzoltan) [inline piker/clearing/_messages.py:187]: ah, so thats where I got it from haha. seems good now

Comment (goodboy) [reply piker/clearing/_messages.py:187]: yah, made #297 as mentioned to addres this.

ideally this sorta thing isn't possible going forward.

please do test if this works if you can though 🏄🏼

Comment (iamzoltan) [reply piker/clearing/_messages.py:187]: tested and works for live orders, not sure if it is also working for dark ones

File: piker/brokers/kraken.py (hunk 1 above)
Comment (iamzoltan) [reply piker/brokers/kraken.py:127]: no doubt. I need to get that up too haha

File: piker/brokers/kraken.py (hunk 2 above)
Comment (iamzoltan) [reply piker/brokers/kraken.py:627]: makes sense. Is this something you want to address in this PR?

Comment (goodboy) [reply piker/brokers/kraken.py:627]: I don't really ever like seeing `except AssertionError:` unless in some kinda wonky testing (and even then).

If you think this works then i'm fine to land it - makes the code a lot more readable for me.

File: piker/brokers/kraken.py (hunk 3 above)
Comment (goodboy) [reply piker/brokers/kraken.py:764]: yeah i think so. doesn't have to be solved here which is why i made #296 to address it more specifically. unfortunately we'll likely have to test with real clearings?

File: piker/brokers/kraken.py [hunk 5]
```diff
@@ -616,42 +617,41 @@ async def handle_order_requests(
...
-                            status='cancelled',
-                            reason='Order cancelled',
-                            broker_details={'name': 'kraken'}
-                        ).dict()
-                    )
-            except AssertionError:
-                log.error(f'Order cancel was not successful')
+            # Check to make sure there was no error returned by
+            # the kraken endpoint. Assert one order was cancelled
+            assert resp['error'] == []
```
Comment (goodboy) [inline piker/brokers/kraken.py:622]: yeah like, normally the way i would do this is to try to get the result and then if one isn't there fail over to error lookup.

maybe i'll change this quick.

**Pull request merged**